deepchem/utils/pytorch_utils.py

```diff
@@ -14,3 +14,53 @@ def get_activation(fn: Union[Callable, str]):
     if isinstance(fn, str):
         return getattr(torch.nn.functional, fn)
     return fn
+
+
+def unsorted_segment_sum(data: torch.Tensor, segment_ids: torch.Tensor,
+                         num_segments: int) -> torch.Tensor:
+    """Computes the sum along segments of a tensor. Analogous to tf.unsorted_segment_sum.
+
+    Parameters
+    ----------
+    data: torch.Tensor
+        A tensor whose segments are to be summed.
+    segment_ids: torch.Tensor
+        The segment indices tensor.
+    num_segments: int
+        The number of segments.
+
+    Returns
+    -------
+    tensor: torch.Tensor
+
+    Examples
+    --------
+    >>> segment_ids = torch.Tensor([0, 1, 0]).to(torch.int64)
+    >>> data = torch.Tensor([[1, 2, 3, 4], [5, 6, 7, 8], [4, 3, 2, 1]])
+    >>> num_segments = 2
+    >>> result = unsorted_segment_sum(data=data,
+                                  segment_ids=segment_ids,
+                                  num_segments=num_segments)
+    >>> result
+    tensor([[5., 5., 5., 5.],
+        [5., 6., 7., 8.]])
+
+    """
+    # length of segment_ids.shape should be 1
+    assert len(segment_ids.shape) == 1
+
+    # Shape of segment_ids should be equal to first dimension of data
+    assert segment_ids.shape[-1] == data.shape[0]
+
+    if len(segment_ids.shape) == 1:
```

Comment:
You can remove this if statement because of the earlier assert

Reply:
Removed